src/reportengine/formattingtools.py

```diff
@@ -27,6 +27,9 @@ def get_nice_name(ns, nsspec, suffix=None):
     (see the ``namespaces`` documentation for more details)"""
     parts = []
     currspec = []
+    prefix = ns.get("filename_prefix", None)
```

Comment:
Is the `filename_prefix` resolved from `ns_dump_description` or by checking the attributes of `environment`? Presumably the environment object is transformed to a dictionary at some point right?

Reply:
Yes. Not proud of that interface. See `ns_dump` below.